File: src/leela/logger.py
#!/usr/bin/python
# -*- coding: utf-8; -*-

# Copyright 2012 Juliano Martinez
# All Rights Reserved.
#
#    Licensed under the Apache License, Version 2.0 (the "License");
#    you may not use this file except in compliance with the License.
#    You may obtain a copy of the License at
#
#        http://www.apache.org/licenses/LICENSE-2.0
#
#    Unless required by applicable law or agreed to in writing, software
#    distributed under the License is distributed on an "AS IS" BASIS,
#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#    See the License for the specific language governing permissions and
#    limitations under the License.
#
# @author: Juliano Martinez

import os.path
import sys
import logging
from logging import DEBUG
from logging import INFO
from logging import WARNING
from logging import ERROR

def set_level(level):
    logging.getLogger().setLevel(level)

def debug(*args, **kwargs):
    logging.debug(*args, **kwargs)

def info(*args, **kwargs):
    logging.info(*args, **kwargs)

def warn(*args, **kwargs):
    logging.warn(*args, **kwargs)

def error(*args, **kwargs):
    logging.error(*args, **kwargs)

def exception(*args, **kwargs):
    logging.exception(*args, **kwargs)

File: src/dmproc/DarkMatter/Logger.hs
-- -*- mode: haskell; -*-
-- All Rights Reserved.
--
--    Licensed under the Apache License, Version 2.0 (the "License");
--    you may not use this file except in compliance with the License.
--    You may obtain a copy of the License at
--
--        http://www.apache.org/licenses/LICENSE-2.0
--
--    Unless required by applicable law or agreed to in writing, software
--    distributed under the License is distributed on an "AS IS" BASIS,
--    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
--    See the License for the specific language governing permissions and
--    limitations under the License.

module DarkMatter.Logger
       ( Priority(DEBUG, INFO, WARNING, ERROR)
       , setlevel
       , debug
       , info
       , warn
       , crit
       )  where

import System.Log
import System.Log.Logger
import System.Time
import System.Locale

myname :: String
myname = "darkmatter"

setlevel :: Priority -> IO ()
setlevel p = updateGlobalLogger myname (setLevel p)

logfmt :: String -> IO String
logfmt s = do { when <- datetime
              ; return ("[darkmatter|" ++ when ++ "] " ++ s)
              }
  where datetime = do { clock    <- getClockTime
                      ; calendar <- toCalendarTime clock
                      ; return (formatCalendarTime defaultTimeLocale "%Y%m%dT%H%M%s%S %Z" calendar)
                      }

debug :: String -> IO ()
debug s = logfmt s >>= debugM myname

info :: String -> IO ()
info s = logfmt s >>= infoM myname

warn :: String -> IO ()
warn s = logfmt s >>= warningM myname

crit :: String -> IO ()
crit s = logfmt s >>= errorM myname

